src/TLuaInterpreter.cpp

```diff
@@ -3250,202 +3250,290 @@ int TLuaInterpreter::setBackgroundImage( lua_State *L )
 
 int TLuaInterpreter::setLabelClickCallback( lua_State *L )
 {
-    string luaSendText="";
-    if( ! lua_isstring( L, 1 ) )
-    {
-        lua_pushstring( L, "setLabelClickCallback: wrong argument type" );
-        lua_error( L );
-        return 1;
-    }
-    else
-    {
-        luaSendText = lua_tostring( L, 1 );
-    }
-    string luaName="";
-    if( ! lua_isstring( L, 2 ) )
-    {
-        lua_pushstring( L, "setLabelClickCallback: wrong argument type" );
-        lua_error( L );
-        return 1;
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if (! pHost) {
+        lua_pushstring(L, tr("setLabelClickCallback: NULL Host pointer - something is wrong!")
+                       .toUtf8().constData());
+        return lua_error(L);
     }
-    else
-    {
-        luaName = lua_tostring( L, 2 );
+
+    QString labelName;
+    if (! lua_isstring(L, 1)) {
+        lua_pushstring(L, tr("setLabelClickCallback: bad argument #1 type (label name as string expected, got %1!)")
```

Comment:
I'm not totally on-board with _not_ coding for it at this point and was retaining the use of the methodology that I started using some (years ?) ago, should it become necessary to rip it all out it can be done but I don't want to do that until I am certain that it is totally irrelevant/harmful.